earth2grid/healpix.py

```diff
@@ -91,10 +96,57 @@ def pad(x: torch.Tensor, padding: int) -> torch.Tensor:
         return healpixpad.HEALPixPadFunction.apply(x.unsqueeze(2), padding).squeeze(2)
 
 
+def _apply_cuhpx_remap(func, x, **kwargs):
+    shape = x.shape
+    x = x.view(-1, 1, shape[-1])
+    nside = npix2nside(x.shape[-1])
+    x = func(x.contiguous(), **kwargs, nside=nside)
+    x = x.contiguous()
+    x = x.view(shape[:-1] + (-1,))
+    return x
+
+
+def npix2nside(npix: int):
+    nside = math.sqrt(npix // 12)
+    return int(nside)
+
+
+def npix2level(npix: int):
+    return nside2level(npix2nside(npix))
+
+
+def nside2level(nside: int):
+    return int(math.log2(nside))
+
+
 class PixelOrder(Enum):
     RING = 0
     NEST = 1
 
+    def reorder_from_cuda(self, x, src: "PixelOrderT"):
```

Comment:
Is it normal for Enum classes to have compute methods like this?

Reply:
it works, and is tested. not public API. ultimately, I will probably refactor PixelOrder to be a class hierarchy.